internal/provider/kubernetes/controller.go

```diff
@@ -1629,11 +1629,6 @@ func (r *gatewayAPIReconciler) hasManagedClass(obj client.Object) bool {
 
 // processParamsRef processes the parametersRef of the provided GatewayClass.
 func (r *gatewayAPIReconciler) processParamsRef(ctx context.Context, gc *gwapiv1b1.GatewayClass, resourceTree *gatewayapi.Resources) error {
-	if gc.Spec.ParametersRef != nil {
-		if !refsEnvoyProxy(gc) {
-			return fmt.Errorf("unsupported parametersRef for gatewayclass %s", gc.Name)
-		}
-	}
```

Comment:
here's the issue with this new code that was added
* its a `processParamsRef` function for a specific `gc` but it fetches all `EnvoyProxies` and then tries to remove the finalizer for unrelated `EnvoyProxies` 
I see another bug on L1675 where we are removing a finalizer from the EnvoyProxy without actually checking if the EnvoyProxy is being referenced by any other GatewayClass or not.


We are very close to our v0.6 release and I prefer reverting https://github.com/envoyproxy/gateway/pull/1534 and adding the feature later when all issues have been addressed

Reply:
This L1675 was actually added to support this case https://github.com/envoyproxy/gateway/pull/1534#issuecomment-1691130263

Reply:
yes, but the way its written it, it can just rm another EnvoyProxy's finalizer which may actually be referenced by a GatewayClass

Reply:
I assumed it was supporting only one gateway class and therefore only one envoy proxy config or am I understanding it wrong?
Finalizer will be only there if it was referenced by a GatewayClass that was accepted.

Reply:
`its a processParamsRef function for a specific gc but it fetches all EnvoyProxies and then tries to remove the finalizer for unrelated EnvoyProxies`
I agree with that, it could definitely be refactored now and have a list taken out of this function

Reply:
the issue is - its a one level check for the EnvoyProxy without backtracking which GatewayClass it belongs to, it may belong to a accepted GatewayClass from another EG controller

Reply:
Hmm, that could be the truth, so we would need to add the check for ControllerName as well

Reply:
ive raised https://github.com/envoyproxy/gateway/pull/2000 for the revert
@cnvergence we really appreciate the hard work on this feature, the provider codebase is already so complex, and there are many edge cases that needed to be handled before we introduce the `Envoyproxy` finalizer which is not a must have yet, so prefer reverting for now

Reply:
I agree, there are a lot of corner cases, I would be happy to help again, after the 0.6.0.

Reply:
thanks for understanding, I actually prefer pausing on this one, until we've cleaned up some of the GC and EP handling in the code before re adding it